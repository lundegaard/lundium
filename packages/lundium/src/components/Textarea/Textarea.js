import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import TextareaControl from '../TextareaControl';
import FormField from '../FormField';
import Text from '../Text';

const Textarea = ({ value, maxCharacters, afterControl, validate = [], ...rest }) => (
	<FormField
		control={TextareaControl}
		kind="textarea"
		value={value}
		validate={validate}
		{...rest}
		afterControl={
			maxCharacters ? (
				<Fragment>
					<Text className="form-group__counter">
						{value.length} / {maxCharacters}
					</Text>
					{afterControl}
				</Fragment>
			) : (
				afterControl
			)
		}
	/>
);

Textarea.propTypes = {
	/** Additional content to render after the control. */
	afterControl: PropTypes.element,
	/** Additional content to render before the control. */
	beforeControl: PropTypes.element,
	/** Field's control component */
	control: PropTypes.elementType.isRequired,
	/** If set, control is disabled */
	disabled: PropTypes.bool,
	/** Node shown in error state. */
	error: PropTypes.node,
	formGroupClassName: PropTypes.string,
	/** @ignore */
	generatedID: PropTypes.string,
	/** Additional props to pass to the FormGroup */
	groupProps: PropTypes.object,
	/** If `true`, input label has floating animation. */
	hasFloatingLabel: PropTypes.bool,
	/** Used to pair the control with a label. If undefined, we use a generated one. */
	id: PropTypes.any,
	/** If `true`, label is absolutely positioned on the right side of the form group. */
	inline: PropTypes.bool,
	/** If `true`, label is hidden. */
	isLabelHidden: PropTypes.bool,
	/** Used to pass a context className to FormGroup (to properly style FormControl and Label) */
	kind: PropTypes.string,
	/** Label value */
	label: PropTypes.node,
	/** Additional props to pass to the Label */
	labelProps: PropTypes.object,
	/** Legend inside text input. */
	legendText: PropTypes.string,
	maxCharacters: PropTypes.number,
	/** Name of the field */
	name: PropTypes.string.isRequired,
	/** If set, control is readOnly */
	readOnly: PropTypes.bool,
	/** If set, control is required */
	required: PropTypes.bool,
	/** Assign a suffix component (e.g. Icon) to Input */
	suffix: PropTypes.node,
	validate: PropTypes.arrayOf(PropTypes.func),
	value: PropTypes.string,
	/** Node shown in warning state. */
	warning: PropTypes.node,
};

export default Textarea;
