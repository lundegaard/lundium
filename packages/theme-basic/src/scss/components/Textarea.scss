@import '../core';

/*
 -  Form control
 -  (<input class="form-control" type="textarea" />)
 */


textarea {
	&.is-invalid {
		padding-inline-end: $input-padding-x;
		background: none;
	}
}
