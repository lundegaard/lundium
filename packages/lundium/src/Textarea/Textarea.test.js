import React from 'react';
import { shallow } from 'enzyme';

import Textarea from './Textarea';

describe('<Textarea />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Textarea {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
