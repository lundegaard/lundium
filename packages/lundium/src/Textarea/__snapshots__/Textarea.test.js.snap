// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(FormField)
  control={[Function]}
  kind="textarea"
  validate={Array []}
/>
`;
